src/test/java/de/numcodex/feasibility_gui_backend/query/broker/direct/DirectBrokerClientCqlTest.java

```diff
@@ -0,0 +1,116 @@
+package de.numcodex.feasibility_gui_backend.query.broker.direct;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+
+import de.numcodex.feasibility_gui_backend.query.QueryMediaType;
+import de.numcodex.feasibility_gui_backend.query.broker.QueryDefinitionNotFoundException;
+import de.numcodex.feasibility_gui_backend.query.broker.QueryNotFoundException;
+import de.numcodex.feasibility_gui_backend.query.broker.SiteNotFoundException;
+import java.io.IOException;
+import org.hl7.fhir.r4.model.Bundle;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Answers;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class DirectBrokerClientCqlTest {
+
+    private static final Long TEST_BACKEND_QUERY_ID = 1L;
+
+    @SuppressWarnings("unused")
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    FhirConnector fhirConnector;
+
+    @InjectMocks
+    DirectBrokerClientCql client;
+
+    @Test
+    void testPublishNonExistingQuery() {
+        assertThrows(QueryNotFoundException.class, () -> client.publishQuery("does-not-exist"));
+    }
+
+    @Test
+    void testPublishExistingQueryWithoutStructuredQueryDefinition() {
```

Comment:
It's the CQL query definition in this case. You can also just write `QueryDefinition` without `Structured`.